### ISSUE TITLE
Exception uninitialized constant KGCL::Operations

### ISSUE THREAD

**Issue by syphax-bouazzouni** [ncbo/bioportal_web_ui#238]
After merging https://github.com/ncbo/bioportal_web_ui/pull/234 in our local code(Agroportal), I realized it does not work in Rails 7. It raises the following exception `NameError: uninitialized constant KGCL::Operations`

The reason is starting from rails 7, there is a new autoloader called `zeitwerk`. And I don't really know why but it can find files in `app/lib/kgcl`

the solution was to add `config.autoload_paths += %W[#{config.root}/app/lib/kgcl]` in  `config/application.rb`

**Comment by jvendetti** [comment]
I merged the pull request to upgrade to Rails 6.1 with zeitwerk autoloading enabled:

```
➜  bioportal_web_ui git:(master) ✗ bin/rails runner 'p Rails.autoloaders.zeitwerk_enabled?'
true
```

I made several follow-up commits that fixed issues when using zeitwerk mode, including the one you listed above (I don't recommend the approach of adding to the autoload paths).

I think the fixes I committed should work with Rails 7. I'll leave this issue open for a bit in the event that you have time to test and report back.
